src/PackageManagement/AzureAutomationTools.PackageManagement.psm1

```diff
@@ -573,6 +573,156 @@ function Add-AatPackageVariable {
+                },
+                [ordered]@{
+                    Name = 'CustomPackage'
+                    Package = 'https://[storageaccountname].blob.core.windows.net/[ContainerName]/CustomPackage.zip'
+                }
+            )
+        }
+
+        $FilePath = Join-Path -Path (Get-AatPackageFolderPath -Modules) -ChildPath "$Name.json"
+        New-Item -Path $FilePath -Force:$Force
+        if ($Content.Count -gt 0) {
+            $Content | ConvertTo-Json | Set-Content -Path $FilePath
+        }
+        else {
+            Set-Content -Path $FilePath -Value '[]'
+        }
+    }
+}
+
+function Add-AatPackageModule {
+    [CmdletBinding(DefaultParameterSetName = 'WithoutPackage')]
+    param (
+        [Parameter(Mandatory = $true,
+                    Position = 2)]
+        [Parameter(ParameterSetName = 'WithoutPackage')]
+        [Parameter(ParameterSetName = 'WithPackage')]
+        [string]
+        $Name,
+
+        [Parameter(Mandatory = $false,
+                    Position = 3)]
+        [Parameter(ParameterSetName = 'WithoutPackage')]
+        [Parameter(ParameterSetName = 'WithPackage')]
+        [string]
+        $Version,
+
+        [Parameter(Mandatory = $true,
+                    Position = 4,
+                    ParameterSetName = 'WithPackage')]
+        [string]
+        $Package
+    )
+
+    dynamicparam{
+        #create a new ParameterAttribute Object
+        $ModuleFileAttribute = New-Object System.Management.Automation.ParameterAttribute
+        $ModuleFileAttribute.Position = 1
+        $ModuleFileAttribute.Mandatory = $true
+        $ModuleFileAttribute.HelpMessage = "Name of the module file to add the module to"
+
+        $ValidateNotNullOrEmptyAttribute = New-Object System.Management.Automation.ValidateNotNullOrEmptyAttribute
+
+        $ValidateSetAttribute = New-Object -TypeName System.Management.Automation.ValidateSetAttribute -ArgumentList (
+            Get-ChildItem -Path (Get-AatPackageFolderPath -Modules) -Filter '*.json' | %{$_.Name -replace '.json', ''}
+        )
+
+        #create an Attributecollection object for the attribute we just created.
+        $AttributeCollection = New-Object -TypeName System.Collections.ObjectModel.Collection[System.Attribute]
+
+        #add our custom attribute
+        $AttributeCollection.Add($ModuleFileAttribute)
+        $AttributeCollection.Add($ValidateSetAttribute)
+        $AttributeCollection.Add($ValidateNotNullOrEmptyAttribute)
+
+        #add our paramater specifying the attribute collection
+        $ModuleFileParam = New-Object System.Management.Automation.RuntimeDefinedParameter('ModuleFile', [string], $AttributeCollection)
+
+        #expose the name of our parameter
+        $ParamDictionary = New-Object System.Management.Automation.RuntimeDefinedParameterDictionary
+        $ParamDictionary.Add('ModuleFile', $ModuleFileParam)
+        return $ParamDictionary
+    }
+
+    begin {
```

Comment:
This is needed as there is a dynamicparam block included. I have removed the unnecessary 'begin' block. The end block is still needed.